modules/optimization/src/optimizationreporters/OptimizationReporter.C

```diff
@@ -12,6 +12,31 @@
 
 registerMooseObject("OptimizationApp", OptimizationReporter);
 
+namespace // unnamed namespace
+{
+void
+copyReporterIntoPetscVector(const std::vector<std::vector<Real> *> reporterVectors,
+                            libMesh::PetscVector<Number> & x)
+{
+  dof_id_type n = 0;
+  for (const auto & data : reporterVectors)
+    for (const auto & val : *data)
+      x.set(n++, val);
+
+  x.close();
+}
+
+void
+copyPetscVectorIntoReporter(const libMesh::PetscVector<Number> & x,
```

Comment:
Make these static member functions in the OptimizationReporter class. Might be useful for other optimizations, so make them public.